diffsims/generators/rotation_list_generators.py

```diff
@@ -159,83 +161,521 @@ def get_grid_around_beam_direction(beam_rotation, resolution, angular_range=(0,
+    # first euler angle: around z = angle between x and x'
+    x_comp = vectors[:, 0]
+    y_comp = vectors[:, 1]
+    norm_proj = np.linalg.norm(vectors[:, :2], axis=1)
+    sign = np.sign(y_comp)
+    # correct for where we have y=0
+    sign[y_comp == 0] = np.sign(x_comp[y_comp == 0])
+    phi2 = sign*np.nan_to_num(np.arccos(x_comp/norm_proj))
+    # phi1 is just 0, rotation around z''
+    phi1 = np.zeros(phi2.shape[0])
+    grid = np.rad2deg(np.vstack([phi1, Phi, phi2]).T)
+    return grid
+
+
+def get_beam_directions_grid(crystal_system, resolution,
+                             mesh="spherified_cube_corner"):
+    """
+    Produces an array of beam directions, within the stereographic
+    triangle of the relevant crystal system. The way the array is constructed
+    is based on different methods of meshing the sphere [1] and can be
+    specified through the `mesh` argument.
+
+    Parameters
+    ----------
+    crystal_system : str
+        Allowed are: 'cubic','hexagonal','trigonal','tetragonal',
+        'orthorhombic','monoclinic','triclinic'
+
+    resolution : float
+        An angle in degrees representing the worst-case angular
+        distance to a first nearest neighbor grid point.
+
+    mesh : str
+        Type of meshing of the sphere that defines how the grid is created.
+        Options are: uv_sphere, normalized_cube, spherified_cube_corner 
+        (default), spherified_cube_edge, and icosahedral.
+
+    Returns
+    -------
+    points_in_cartesians : np.array (N,3)
+        Rows are x,y,z where z is the 001 pole direction.
+    """
+    if mesh == "uv_sphere":
+        points_in_cartesians = get_uv_sphere_mesh_vertices(resolution)
+    elif mesh == "normalized_cube":
+        # special case: hexagon is a very small slice and 001 point can
+        # be isolated. Hence we increase resolution to ensure minimum angle.
+        if crystal_system == "hexagonal":
+            resolution = np.rad2deg(
+                    np.arctan(np.tan(np.deg2rad(resolution))/np.sqrt(2)))
+        points_in_cartesians = get_cube_mesh_vertices(resolution,
+                                                      grid_type="normalized")
+
+    elif mesh == "spherified_cube_edge":
+        # special case: hexagon is a very small slice and 001 point can
+        # be isolated.  Hence we increase resolution to ensure minimum angle.
+        if crystal_system == "hexagonal":
+            resolution = np.rad2deg(
+                    np.arctan(np.tan(np.deg2rad(resolution))/np.sqrt(2)))
+        points_in_cartesians = get_cube_mesh_vertices(resolution,
+                                                      grid_type="spherified_edge")
+    elif mesh == "spherified_cube_corner":
+        points_in_cartesians = get_cube_mesh_vertices(resolution,
+                                                      grid_type="spherified_corner")
+    elif mesh == "icosahedral":
+        points_in_cartesians = get_icosahedral_mesh_vertices(resolution)
```

Comment:
can you reorder the logic here to go `if mesh` for the three that are one liners and then:

```
elif mesh == "normalized_cube" or "spherified_cube_edge":
      if crystal_system="hexagonal":
                     # stuff
      if mesh == "normalized_cube":
              points_in_cartesians = ...
     else:
             points_in_cartesians = ...
```